File: Server/src/main/java/core/game/content/dialogue/LidioDialogue.java
package core.game.content.dialogue;

import core.game.node.entity.npc.NPC;
import core.plugin.Initializable;
import core.game.node.entity.player.Player;

/**
 * Represents the dialogue plugin used for the lidio npc.
 * @author 'Vexia
 * @version 1.0
 */
@Initializable
public final class LidioDialogue extends DialoguePlugin {

	/**
	 * Constructs a new {@code LidioDialogue} {@code Object}.
	 */
	public LidioDialogue() {
		/**
		 * empty.
		 */
	}

	/**
	 * Constructs a new {@code LidioDialogue} {@code Object}.
	 * @param player the player.
	 */
	public LidioDialogue(Player player) {
		super(player);
	}

	@Override
	public DialoguePlugin newInstance(Player player) {
		return new LidioDialogue(player);
	}

	@Override
	public boolean open(Object... args) {
		npc = (NPC) args[0];
		interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "Greetings, warrior, how can I fill your stomach today?");
		stage = 0;
		return true;
	}

	@Override
	public boolean handle(int interfaceId, int buttonId) {
		switch (stage) {
		case 0:
			interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "With food preferrable.");
			stage = 1;
			break;
		case 1:
			end();
			npc.openShop(player);
			break;
		}
		return true;
	}

	@Override
	public int[] getIds() {
		return new int[] { 4293 };
	}
}

File: Server/src/main/content/region/asgarnia/burthorpe/dialogue/LidioDialogue.kt
package content.region.asgarnia.burthorpe.dialogue

import core.api.openDialogue
import core.api.openNpcShop
import core.game.dialogue.DialogueBuilder
import core.game.dialogue.DialogueBuilderFile
import core.game.dialogue.DialoguePlugin
import core.game.dialogue.FacialExpression
import core.game.node.entity.player.Player
import core.plugin.Initializable
import org.rs09.consts.NPCs

@Initializable
class LidioDialogue(player: Player? = null) : DialoguePlugin(player) {

    override fun handle(interfaceId: Int, buttonId: Int): Boolean {
        openDialogue(player, LidioDialogueFile(), npc)
        return true
    }

    override fun newInstance(player: Player?): DialoguePlugin {
        return LidioDialogue(player)
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.LIDIO_4293)
    }
}

class LidioDialogueFile : DialogueBuilderFile() {

    override fun create(b: DialogueBuilder) {
        b.defaultDialogue().npcl(FacialExpression.ASKING,
            "Greetings, warrior, how can I fill your stomach today?"
        ).playerl(FacialExpression.NEUTRAL,
                "With food preferrable."
            ).endWith { _, player ->
                openNpcShop(player, NPCs.LIDIO_4293)
                end()
            }
    }
}